shared/src/main/scala/mlscript/TypeSimplifier.scala

```diff
@@ -92,7 +92,7 @@ trait TypeSimplifier { self: Typer =>
       case ProvType(ty) => process(ty, parent, canDistribForall = canDistribForall)
       
       case tr @ TypeRef(defn, targs) if builtinTypes.contains(defn) && tr.canExpand =>
-        process(tr.expandOrCrash, parent)
+        process(tr.expandOrCrash(true), parent)
```

Comment:
This will be incorrect if the type has type parameters.
```suggestion
        assert(tr.targs.isEmpty)
        process(tr.expandOrCrash(true), parent)
```